### PR TITLE
feat(osu!ppy): add language file & add strings to general file

### PR DESCRIPTION
**Pull request by DarkVillager**
feat(osu!ppy): add language file & add strings to general file
Goes with https://github.com/PreMiD/Presences/pull/7588

### CODE REVIEW

File: src/Presence/osu!ppy.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		"description": "Example: Performance Rankings (for All) [osu!]",
+		"message": "for {0} {1}"
+	},
+	"osu!ppy.performanceRankings": {
+		"description": "Displayed when viewing the performance rankings.",
+		"message": "Performance Rankings"
+	},
+	"osu!ppy.readingNews": {
+		"description": "Displayed when reading a news page.",
+		"message": "Reading News:"
```
Comment (Bas950) [inline src/Presence/osu!ppy.json:36]: would probably be good as a general string.

File: src/Presence/osu!ppy.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+		"description": "Example: Score Rankings [osu!]",
+		"message": "Score Rankings {0}"
+	},
+	"osu!ppy.spotlights": {
+		"description": "Example: The osu! Spotlights (All)",
+		"message": "The {0} Spotlights {1}"
+	},
+	"osu!ppy.support": {
+		"description": "Displayed when viewing the support page.",
+		"message": "Support"
```
Comment (Bas950) [inline src/Presence/osu!ppy.json:48]: general.support ?

File: src/Presence/osu!ppy.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+		"message": "View News Article"
+	},
+	"osu!ppy.viewingNews": {
+		"description": "Displayed when viewing all osu! news.",
+		"message": "Viewing News Articles"
+	},
+	"osu!ppy.wikiMainPage": {
+		"description": "Displayed when viewing the wiki main page.",
+		"message": "Wiki Main Page"
+	}
```
Comment (Bas950) [inline src/Presence/osu!ppy.json:61]: Could all be good as general strings

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: src/Presence/osu!ppy.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
+{
+	"osu!ppy.beatMapListing": {
+		"description": "Displayed when viewing all beatmaps.",
+		"message": "Viewing Beatmap Listings"
+	},
+	"osu!ppy.beatMapLooking": {
+		"description": "Displayed when looking at a beatmap.",
+		"message": "Looking at Beatmap"
```
[the remark this answers is not in the document]
Comment (DarkVillager) [reply src/Presence/osu!ppy.json:8]: shows the title.

File: src/Presence/osu!ppy.json (hunk 2 above)
Comment (DarkVillager) [reply src/Presence/osu!ppy.json:48]: This will get changed.

File: src/Presence/osu!ppy.json (hunk 3 above)
Comment (DarkVillager) [reply src/Presence/osu!ppy.json:61]: Ok.

File: src/Presence/general.json [hunk 5]
```diff
@@ -19,6 +19,10 @@
     "description": "Used as the title for the button in Discord.",
     "message": "Read Article"
   },
+  "general.buttonReadNews": {
+    "description": "Used as the title for the button in Discord.",
+    "message": "Read News"
+  },
```
Comment (Bas950) [inline src/Presence/general.json:25]: Can't you just use Read Article?

File: src/Presence/general.json [hunk 6]
```diff
@@ -207,6 +215,10 @@
     "description": "",
     "message": "Reading DMs from:"
   },
+  "general.readingNews": {
+    "description": "Displayed when reading a news page.",
+    "message": "Reading news about:"
+  },
```
Comment (Bas950) [inline src/Presence/general.json:221]: Ditto

File: src/Presence/osu!ppy.json [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    "message": "Viewing kudosu rankings"
+  },
+  "osu!ppy.buttonViewBeatmap": {
+    "description": "Displayed for a button to redirect to the beatmap page.",
+    "message": "View Beatmap"
+  },
+  "osu!ppy.countryRankings": {
+    "description": "Example: Country Rankings [osu!]",
+    "message": "Country rankings {0}"
+  },
```
Comment (Bas950) [inline src/Presence/osu!ppy.json:25]: so what does 0 get replaced with?

File: src/Presence/osu!ppy.json [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  "osu!ppy.for": {
+    "description": "Example: Performance Rankings (for All) [osu!]",
+    "message": "for {0} {1}"
+  },
```
Comment (Bas950) [inline src/Presence/osu!ppy.json:29]: Same here the description just isn't enough.

File: src/Presence/osu!ppy.json [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    "message": "Performance rankings"
+  },
+  "osu!ppy.readingNews": {
+    "description": "Displayed when reading a news page.",
+    "message": "Reading news:"
+  },
+  "osu!ppy.scoreRankings": {
+    "description": "Example: Score Rankings [osu!]",
+    "message": "Score rankings {0}"
+  },
```
Comment (Bas950) [inline src/Presence/osu!ppy.json:41]: ditto

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: src/Presence/general.json (hunk 5 above)
Comment (DarkVillager) [reply src/Presence/general.json:25]: You be the judge of that. Example url: [https://osu.ppy.sh/home/news/2023-08-29-changes-to-the-mappers-guild](https://osu.ppy.sh/home/news/2023-08-29-changes-to-the-mappers-guild)

Comment (Bas950) [reply src/Presence/general.json:25]: I think you can

File: src/Presence/osu!ppy.json (hunk 7 above)
Comment (Bas950) [reply src/Presence/osu!ppy.json:25]: Please update the description to include information of the variable

File: src/Presence/osu!ppy.json (hunk 8 above)
Comment (Bas950) [reply src/Presence/osu!ppy.json:29]: Please update the description to include information of the variable

File: src/Presence/osu!ppy.json (hunk 9 above)
Comment (Bas950) [reply src/Presence/osu!ppy.json:41]: Please update the description to include information of the variable

File: src/Presence/osu!ppy.json [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+		"description": "Displayed for a button to redirect to the beatmap page.",
+		"message": "View Beatmap"
+	},
+	"osu!ppy.countryRankings": {
+		"description": "Country rankings {0} = gamemode. E.g. Country Rankings [osu!]",
+		"message": "Country rankings {0}"
+	},
+	"osu!ppy.for": {
+		"description": "{0} = country/for All, {1} = gamemode. E.g. Performance Rankings (for All) [osu!]",
+		"message": "for {0} {1}"
```
Comment (Bas950) [inline src/Presence/osu!ppy.json:28]: Description states `(for All) [osu!]`
Message states `for {0} {1}`
Shouldn't there be rounded brackets around `for {0}` ?

File: src/Presence/osu!ppy.json [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+		"description": "Displayed when viewing the performance rankings.",
+		"message": "Performance rankings"
+	},
+	"osu!ppy.scoreRankings": {
+		"description": "{0} = gamemode. E.g. Score Rankings [osu!]",
+		"message": "Score rankings {0}"
+	},
+	"osu!ppy.spotlights": {
+		"description": "{0} = gamemode, {1} = All/Friends. E.g. The osu! Spotlights (All)",
+		"message": "The {0} spotlights {1}"
```
Comment (Bas950) [inline src/Presence/osu!ppy.json:40]: Ditto

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]

**Review by Timeraa — APPROVED** [review]

**Pull request merged**